### PR TITLE
add argument checks for file routing

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Comment by SimonCropp** [comment]
@SzymonPobiega thoughts?

File: src/NServiceBus.Core/Routing/FileBasedDynamicRouting/FileRoutingTableSettings.cs [hunk 1]
```diff
@@ -20,20 +20,36 @@ public FileRoutingTableSettings(SettingsHolder settings)
...
 
         /// <summary>
         /// Specifies the maximum number of attempts to load contents of a file before logging an error.
+        /// The default value is 10 attempts.
+        /// Valid values must be at least 1.
         /// </summary>
         /// <param name="maxLoadAttempts">Max load attempts.</param>
         public FileRoutingTableSettings MaxLoadAttempts(int maxLoadAttempts)
         {
+            if (maxLoadAttempts < 1)
```
Comment (timbussmann) [inline src/NServiceBus.Core/Routing/FileBasedDynamicRouting/FileRoutingTableSettings.cs]: you could also use `Guard.AgainstNegativeAndZero` I think?

**Comment by timbussmann** [comment]
I like the checks :+1:

**Comment by danielmarbach** [comment]
:+1: I don't think we need to address @timbussmann comment since it is consistent in this file

**Pull request merged**